extendr-macros/src/lib.rs

```diff
@@ -168,3 +169,67 @@ pub fn R(item: TokenStream) -> TokenStream {
 pub fn Rraw(item: TokenStream) -> TokenStream {
     R::R(item.into(), false).into()
 }
+
+/// Derives an implementation of `TryFrom<Robj> for Struct` and `TryFrom<&Robj> for Struct` on this struct.
+///
+/// This allows you to convert any R object supporting the `$` operator (generally a list or an
+/// environment) into your struct, as long as the corresponding fields on the R object are
+/// of a compatible type to those on the Rust struct.
+///
+/// # Examples
+/// ```ignore
+/// use extendr_api::prelude::*;
+/// use extendr_macros::TryFromRobj;
+///
+/// #[derive(TryFromRobj)]
+/// struct Foo {
+///     a: usize,
+///     b: String
+/// }
+/// ```
+/// Then, you can create an instance of your struct from an R expression:
+/// ```ignore
+/// let foo: Foo = R!("list(a = 5, b = 'bar')").unwrap().try_into().unwrap();
+/// ```
```

Comment:
I think we have never used 'second person' in comments/examples. It might be better if it is 'us'/'ours' (if directed towards us the maintainers) or perhaps passive (like the next docs block) -- neutral and directed to consumers?